aes/decryptor.go

```diff
@@ -0,0 +1,122 @@
+package aes
+
+import (
+	cryptoaes "crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"errors"
+	"fmt"
+	"regexp"
+	"strconv"
+)
+
+type EncryptedValue struct {
+	data     []byte
+	iv       []byte
+	tag      []byte
+	datatype string
+}
+
+func parse(value string) (*EncryptedValue, error) {
+	re := regexp.MustCompile(`^ENC\[AES256_GCM,data:(.+),iv:(.+),tag:(.+),type:(.+)\]`)
```

Comment:
protip: move that out of the function to avoid rebuilding it all the time

```go
var encre = regexp.MustCompile(`^ENC\[AES256_GCM,data:(.+),iv:(.+),tag:(.+),type:(.+)\]`)
func parse(value string) (*EncryptedValue, error) {
```